packager/hls/base/simple_hls_notifier.cc

```diff
@@ -373,8 +373,27 @@ bool SimpleHlsNotifier::NotifyNewSegment(uint32_t stream_id,
   const std::string& segment_url =
       GenerateSegmentUrl(segment_name, hls_params().base_url,
                          master_playlist_dir_, media_playlist->file_name());
+
+  // Check if we need to set reference_time_
+  // This need to be adjusted on discontinuity
+  if (hls_params().add_ext_x_program_date_time) {
+    if (reference_time_.is_null() || start_time < media_playlist->LastSegmentStartTime()) {
```

Comment:
I like the idea of resetting the reference time when there are discontinuities!

There is a minor problem here though. Let's assume that there are two streams. The reference time will be reset twice for every discontinuity as the discontinuity exists in both streams.

Can you we always use one stream as reference (except for the initial case)?

The code can be structured as something like:

```
  if (hls_params().add_ext_x_program_date_time) {
    if (reference_time_.is_null()) {
      reference_time_ = CalculateReferenceTime(start_time);
      reference_stream_id_ = stream_id;
    } else if (stream_id == reference_stream_id_ && media_playlist->IsDiscontinuity(start_time)) {
      reference_time_ = CalculateReferenceTime(start_time);
    }
    media_playlist->SetReferenceTime(reference_time_);
  }
```

`IsDiscontinuity(start_time)` is a MediaPlaylist function which wraps `start_time < media_playlist->LastSegmentStartTime()`, as it is easier to understand and also easier to maintain in the future.